### PR TITLE
provision phpcs + wp sniffs via composer

### PR DESCRIPTION
**Pull request by cfoellmann**
Relates to #922 

Introduce composer installation of phpcs and the sniffs for WP and keep it in the right place.
Removes 2 git operations per provision.

### CODE REVIEW

**Comment by travisseitler** [comment]
(Anecdotal, but I merged this patch on my machine and it appears to also address the phpcs provisioning issue at #1179.)

**Comment by thebrandonallen** [comment]
`"squizlabs/php_codesniffer": "*"` will need to be changed to `"squizlabs/php_codesniffer": "2.*"`, since the WordPress Coding Standards sniffs are not compatible with 3.0.

**Comment by cfoellmann** [comment]
@thebrandonallen made the change to phpcs 2.x

File: provision/phpcs/composer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "VVV/phpcs",
+    "type": "project",
+    "require": {
+        "oomphinc/composer-installers-extender": "^1.1",
+        "squizlabs/php_codesniffer": "2.*",
```
Comment (TangRufus) [inline provision/phpcs/composer.json]: I think requiring just `wp-coding-standards/wpcs` is enough, as `wpcs` requires `squizlabs/php_codesniffer`

Comment (cfoellmann) [inline provision/phpcs/composer.json]: you are right. Fix for the PR is coming

**Comment by cfoellmann** [comment]
squashed everything (not best practice, i know).
Works on my install. review would be nice.

**Comment by tomjn** [comment]
This PR broke `PATH` access to `phpcs` and diverged old and new machines by changing the folders the executables are saved to:

old: `/srv/www/phpcs/scripts/phpcs`
new: `/srv/www/phpcs/bin/phpcs`

**Pull request merged**